hydradx/tests/test_liquidations.py

```diff
@@ -737,279 +754,298 @@ def test_liquidate_against_omnipool_fuzz(collateral_amt1: float, ratio1: float,
     )
 
-    state = GlobalState(
-        agents={liquidating_agent_id: Agent()},
-        pools={pool_id: omnipool},
-        money_market=mm
+    collateral = {'USDT': final_price['DOT'], 'DOT': 1}
+    mm.prices = final_price
+    hdx_debt_half_liquidation_point = mm.value_assets(collateral) / final_price['HDX'] * default_liquidation_threshold
+    hdx_debt_full_liquidation_point = hdx_debt_half_liquidation_point / 0.95
+    # liquidate half
+    cdp1 = CDP(
+        {'HDX': hdx_debt_full_liquidation_point - 0.00001},
+        collateral
     )
-
-    evolve_fn = update_prices_and_process(pool_id, liquidating_agent_id, price_list, "USDT")
-    evolve_fn(state)
-
-    if cdp1.debt_amt != 0:
-        raise ValueError("CDP1 should be fully liquidated")
-    if cdp2.debt_amt != 1:
-        raise ValueError("CDP2 should not be liquidated")
-    if cdp3.debt_amt != 0:
-        raise ValueError("CDP3 should be fully liquidated")
-    if mm.get_oracle_price("HDX", "USDT") != 0.03:
-        raise ValueError("Oracle price for HDX should be 0.03")
-    if mm.get_oracle_price("DOT", "USDT") != 5:
-        raise ValueError("Oracle price for DOT should be 5")
-    if mm.get_oracle_price("HDX", "DOT") != 0.03 / 5:
-        raise ValueError("Oracle price for HDX/DOT should be 0.03 / 5 ")
+    # liquidate all
+    cdp2 = CDP(
+        {'HDX': hdx_debt_full_liquidation_point + 0.00001},
+        collateral
+    )
+    # liquidate half
+    cdp3 = CDP(
+        {'HDX': hdx_debt_half_liquidation_point + 0.00001},
+        collateral
+    )
+    # liquidate non
+    cdp4 = CDP(
+        {'HDX': hdx_debt_half_liquidation_point - 0.00001},
+        collateral
+    )
+    mm.prices = initial_price
+    mm.cdps = [cdp1, cdp2, cdp3, cdp4]
+    final_state = run.run(initial_state, 1)[-1]
+    final_mm = final_state.money_market
+    final_cdp1 = final_mm.cdps[0]
+    final_cdp2 = final_mm.cdps[1]
+    final_cdp3 = final_mm.cdps[2]
+    final_cdp4 = final_mm.cdps[3]
+    if final_cdp2.collateral['DOT'] > 0:
+        raise ValueError("CDP2 DOT should be fully liquidated")
+    if (final_mm.value_assets({'DOT': cdp1.collateral['DOT'] - final_cdp1.collateral['DOT']})
+            / final_mm.value_assets({'HDX': final_cdp1.debt['HDX']}) - 1
+            != pytest.approx(final_mm.liquidation_bonus[('DOT', 'HDX')], rel=1e-12)
+    ):
+        raise ValueError("CDP1 DOT should be liquidated at half the value of HDX debt + liquidation bonus")
+    if (final_mm.value_assets({'DOT': cdp3.collateral['DOT'] - final_cdp3.collateral['DOT']})
+            / final_mm.value_assets({'HDX': final_cdp3.debt['HDX']}) - 1
+            != pytest.approx(final_mm.liquidation_bonus[('DOT', 'HDX')], rel=1e-12)
+    ):
+        raise ValueError("CDP3 should be half liquidated just like CDP1")
+    if final_cdp4.debt != cdp4.debt or final_cdp4.collateral != cdp4.collateral:
+        raise ValueError("CDP4 should not be liquidated")
+    er = 1
```

Comment:
I think it is worth carefully testing cdp_liquidation_threshold, now that we have implemented multi-collateral logic. I would suggest testing a few hardcoded cases where you can work through the expected results in a spreadsheet, and hard code them into the test. Also test get_health_factor on the same examples.